TicTacToe/TicTacToe/Board.cs

```diff
@@ -1,18 +1,38 @@
 ﻿using System;
+using System.Collections.Generic;
 
 namespace TicTacToe {
     
     public class Board {
 
+        private List<string> availableSpaces = new List<string>();
         private string[] gameBoard = {"0", "1", "2", "3", "4", "5", "6", "7", "8"};
         private string currentMarker = "X";
 
         public string[] GameBoard {
             get { return gameBoard; }
         }
 
+        public List<string> GetAvailableSpaces {
+            get { return availableSpaces; }
+        }
+
+        public void CreateNewBoard() {
+            AvailableSpaces();
+        }
+
         public void UpdateBoard(int move) {
             gameBoard[move] = currentMarker;
+            AvailableSpaces();
+        }
+
+        private void AvailableSpaces() {
+            availableSpaces.Clear();
+            foreach (string cell in gameBoard) {
+                if (Int32.TryParse(cell, out int number)) {
+                    availableSpaces.Add(cell);
+                }
```

Comment:
Since your board is stored as a `string[]`, i.e., effectively a list, you could replace iteration plus conditional with filtering here. Investigate [`Where`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.where?view=netcore-2.0#System_Linq_Enumerable_Where__1_System_Collections_Generic_IEnumerable___0__System_Func___0_System_Boolean__) and related methods if this appeals to you right now.